src/main/java/core/basesyntax/dao/ReportExporterDaoImpl.java

```
@@ -0,0 +1,27 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.exeptions.FileWriteException;
+import core.basesyntax.exeptions.IncorrectFileNameException;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class ReportExporterDaoImpl implements ReportExporterDao {
+    private final String fileName;
+
+    public ReportExporterDaoImpl(String fileName) {
+        if (fileName == null || fileName.isBlank()) {
+            throw new IncorrectFileNameException("File name is incorrect " + fileName);
+        }
+        this.fileName = fileName;
+    }
+
+    @Override
+    public void writeTheReport(String report) {
+        try (FileWriter fileWriter = new FileWriter(fileName)) {
+            fileWriter.write(report);
+        } catch (IOException e) {
+            throw new FileWriteException("Error writing report to " + fileName
+                    + ": " + e.getMessage());
```

The catch block rethrows a FileWriteException but does not pass the original IOException as the cause. This violates the checklist item: "Don't ignore exceptions in try-catch blocks... Let's rethrow a RuntimeException with an **informative** message and exception object."  The FileWriteException class currently has only a (String) constructor, so the original exception is lost. Please add a constructor to FileWriteException that accepts (String, Throwable) and call it here, for example: `throw new FileWriteException("Error writing report to " + fileName, e);` This preserves the cause and stack trace. See this class for location.